modules/vectra.py

```diff
@@ -466,17 +477,11 @@ def update_rule(self, rule_id=None, name=None, append=False, **kwargs):
         :param remote1_dns: destination hostnames to triage
         :param remote1_port: destination ports to  triage
         """
-        if not rule_id and not name:
-            raise ValueError("rule name or id must be provided")
 
-        id = self.get_rules(name=name)['id'] if name else rule_id
-        rule = self.get_rules(rule_id=id).json()
-
-        valid_keys = ['ip', 'host', 'sensor_luid', 'remote1_ip', 'remote1_dns', 'remote1_port']
+        rule = self.get_rule_by_id(rule_id=rule_id).json()
+        id = rule['id']
 
         for k, v in kwargs.items():
-            if k not in valid_keys:
```

Comment:
Why should we remove the param validation here?

Reply:
I think this was done because the list isn't actually exhaustive and you could use other parameters. The rest is then laziness; as I didn't have the exhaustive list, I just removed the check